website_rentals/helpers/scheduling.py

```diff
@@ -148,10 +172,101 @@ def get_rental_hourly_timeslots(self, product, start_date, stop_date=None):
         if not stop_times:
             return
 
-        return {
-            "start": list(map(_format_timeslot_time, start_times)),
-            "stop": list(map(_format_timeslot_time, stop_times)),
-        }
+        # Find all possible overlapping reservations
+        overlapping_reservations = self.env['sale.rental.schedule'].search([
+            ('report_line_status', 'in', ['reserved', 'pickedup']),
+            ('product_id', 'in', product.product_tmpl_id.product_variant_ids.ids if product.product_tmpl_id.rental_check_availability_on_all_products else [product.id]),
+            '|',
+            '&', ('pickup_date', '>=', start_date), ('pickup_date', '<=', stop_date + timedelta(days=1)),
+            '&', ('return_date', '>=', start_date), ('return_date', '<=', stop_date + timedelta(days=1)),
+        ])
+
+        # Check for overlaps when the duration is over more than one day. If there are too many overlaps, return no available slots
+        if not is_same_day:
+            if include_start:
+                overlaps = overlapping_reservations.filtered(
+                    lambda r: start_date + timedelta(days=1) <= r.pickup_date < stop_date or start_date + timedelta(days=1) <= r.return_date < stop_date)
+            else:
+                overlaps = overlapping_reservations.filtered(lambda r: start_date <= r.pickup_date < stop_date or start_date <= r.return_date < stop_date)
+
+            if len(overlaps) + quantity > total_units:
+                return {
+                    'start': [],
+                    'stop': []
+                }
+
+        # Search for stop times to remove because they are not available because of overlapping bookings
+        remove_stop_times = []
+        remove_all_following = False
+        for stop_time in stop_times:
+            if not is_same_day:
+                stop_datetime = stop_date.replace(hour=int(stop_time))
+                if include_start:
+                    overlaps = overlapping_reservations.filtered(lambda r: stop_date <= r.pickup_date.astimezone(pytz.timezone(self.env.user.tz or 'Europe/Brussels')).replace(
+                        tzinfo=None) <= stop_datetime or stop_date <= r.return_date.astimezone(pytz.timezone(self.env.user.tz or 'Europe/Brussels')).replace(
```

Comment:
@wardm95 let's try to shift away from hardcoded timezones. Can you try getting the tz from the cookies? 
```
request.httprequest.cookies.get('tz') 
```
or for geoip something like this:
```
request.session.geoip.get('country_code')
```
Ref https://pythonhosted.org/python-geoip/